### PR TITLE
Fix vscode extension

### LINKED ISSUE #1810: VS Code extension does not provide all recent items
# Extension – [Visual Studio Code Recent Projects](https://github.com/raycast/extensions/tree/main/extensions/visual-studio-code-recent-projects)

Author: @thomaspaulmann

## Description

The changes in https://github.com/raycast/extensions/pull/1630 changed to loading the recent VS Code projects from the storage.json path which is what is used to build the VS Code recent projects in the menu.  However, this is not the full list that you get when using the `File: Open recent` command via the command palette.

This is problematic since it only shows the most recent 10 projects rather than the full list.

I made a branch to revert to using the sqlite DB as the source of truth which fixes this issue. 
 https://github.com/raycast/extensions/compare/main...mskelton:fix-vscode-extension

## Steps To Reproduce

1. Open the recent projects command in Raycast

## The current behavior

Notice that only the last 10 projects are listed.

## The expected behavior

All recent projects are listed.

### Raycast version

Version: 1.35.1

### PR DESCRIPTION
**Pull request by mskelton**
## Description

Fixes #1810 

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by macbookandrew** [comment]
This isn’t working for workspace files…seems that perhaps VS Code changed the JSON structure leading to the change in #1630 ?

Here’s the current structure:

```jsonc
{
    "entries": [
        {
            "workspace": {
                "id": "b7fc2276cf54423863a79ff7d46d9714",
                "configPath": "file:///Users/andrewminion/Documents/VS%20Code%20workspaces/Prime%20Therapeutics.code-workspace"
            }
        },
// etc.
```

**Comment by mskelton** [comment]
@macbookandrew Thanks, I've updated my PR to include the fix for workspaces.

**Comment by macbookandrew** [comment]
Awesome…this works great for me!

File: extensions/visual-studio-code-recent-projects/src/db.ts [hunk 1]
```diff
@@ -36,66 +24,13 @@ type QueryResult = {
 }[];
 
 export async function getRecentEntries(): Promise<EntryLike[]> {
-  // VS Code version < 1.64.0
-  if (existsSync(LEGACY_STORAGE_PATH)) {
-    const json = JSON.parse(await readFile(LEGACY_STORAGE_PATH, "utf8"));
-    if (json.openedPathsList) {
-      return json.openedPathsList.entries;
-    }
```
[the remark this answers is not in the document]
Comment (thomaspaulmann) [reply extensions/visual-studio-code-recent-projects/src/db.ts:44]: Seeing @macbookandrew's comment seemst that VS Code also changed the JSON structure. So let's drop support. If some bug report pops up, we can address it but might need to handle then also different JSON structures.

**Pull request merged**